src/components/Identity/createIdSyncs.js

```diff
@@ -91,6 +91,14 @@ const createProcessor = (config, logger, cookieJar) => destinations => {
       ).toString(36)
     );
 
+    if (result.failed.length) {
+      const failedIds = result.failed.map(idSync => idSync.id);
+
+      return Promise.reject(
+        new Error(`The following id sync ids failed: ${failedIds.join(", ")}.`)
+      );
```

Comment:
You can just throw the error rather than returning a rejected promise.
I would recommend capitalizing ID here: `The following ID sync IDs failed: ${failedIds.join(", ")}.`